File: common/src/main/java/dev/number6/db/port/DatabasePort.java
package dev.number6.db.port;

import dev.number6.comprehend.results.PresentableEntityResults;
import dev.number6.comprehend.results.PresentableKeyPhrasesResults;
import dev.number6.comprehend.results.PresentableSentimentResults;

import java.time.LocalDate;
import java.util.Collection;

public interface DatabasePort {

    void createNewSummaryForChannels(Collection<String> channelNames, LocalDate comprehensionDate);

    void save(PresentableSentimentResults results);

    void save(PresentableEntityResults results);

    void save(PresentableKeyPhrasesResults results);
}

File: common/src/main/kotlin/dev/number6/db/port/DatabasePort.kt
package dev.number6.db.port

import dev.number6.comprehend.results.PresentableEntityResults
import dev.number6.comprehend.results.PresentableKeyPhrasesResults
import dev.number6.comprehend.results.PresentableSentimentResults
import java.time.LocalDate

interface DatabasePort {
    fun createNewSummaryForChannels(channelNames: Collection<String>, comprehensionDate: LocalDate)
    fun save(results: PresentableSentimentResults)
    fun save(results: PresentableEntityResults)
    fun save(results: PresentableKeyPhrasesResults)
}